endpoints/test/test_live_auth.py

```diff
@@ -0,0 +1,209 @@
+        some_path = os.path.dirname(some_path)
+    return some_path
+
+PKGDIR = _find_setup_py(TESTDIR)
+
+@pytest.fixture(scope='session')
+def integration_project_id():
+    if 'INTEGRATION_PROJECT_ID' not in os.environ:
+        raise KeyError('INTEGRATION_PROJECT_ID required in environment. Set it to the appropriate project id.')
+    return os.environ['INTEGRATION_PROJECT_ID']
+
+@pytest.fixture(scope='session')
+def service_account_keyfile():
+    if 'SERVICE_ACCOUNT_KEYFILE' not in os.environ:
+        raise KeyError('SERVICE_ACCOUNT_KEYFILE required in environment. Set it to the path to the service account key.')
+    value = os.environ['SERVICE_ACCOUNT_KEYFILE']
+    if not os.path.isfile(value):
+        raise ValueError('SERVICE_ACCOUNT_KEYFILE must point to a file containing the service account key.')
+    return value
+
+@pytest.fixture(scope='session')
+def api_key():
+    if 'PROJECT_API_KEY' not in os.environ:
+        raise KeyError('PROJECT_API_KEY required in environment. Set it to a valid api key for the specified project.')
+    return os.environ['PROJECT_API_KEY']
+
+@pytest.fixture(scope='session')
+def gcloud_driver_module(request):
+    """This fixture provides the gcloud test driver. It is not normally installable, since it lacks a setup.py"""
+    cache_key = 'live_auth/driver_zip'
+    driver_zip_data = request.config.cache.get(cache_key, None)
+    if driver_zip_data is None:
+        url = "https://github.com/GoogleCloudPlatform/cloudsdk-test-driver/archive/master.zip"
+        driver_zip_data = requests.get(url).content
+        request.config.cache.set(cache_key, base64.b64encode(driver_zip_data))
+    else:
+        driver_zip_data = base64.b64decode(driver_zip_data)
+    extract_path = tempfile.mkdtemp()
+    with zipfile.ZipFile(cStringIO.StringIO(driver_zip_data)) as driver_zip:
+        driver_zip.extractall(path=extract_path)
+    # have to rename the subfolder
+    os.rename(os.path.join(extract_path, 'cloudsdk-test-driver-master'), os.path.join(extract_path, 'cloudsdk_test_driver'))
+    sys.path.append(extract_path)
+    driver_module = importlib.import_module('cloudsdk_test_driver.driver')
+    yield driver_module
+    sys.path.pop()
+    shutil.rmtree(extract_path)
+
+@pytest.fixture(scope='session')
+def gcloud_driver(gcloud_driver_module):
+    with gcloud_driver_module.Manager(additional_components=['app-engine-python']):
+        yield gcloud_driver_module
+
+@pytest.fixture(scope='session')
+def gcloud_sdk(gcloud_driver, integration_project_id, service_account_keyfile):
+    return gcloud_driver.SDKFromArgs(project=integration_project_id, service_account_keyfile=service_account_keyfile)
+
+class TestAppMaker(object):
+    def __init__(self):
+        self.cleanup_path = tempfile.mkdtemp()
+        self.app_path = os.path.join(self.cleanup_path, 'app')
+
+    def cleanup(self):
+        shutil.rmtree(self.cleanup_path)
+
+    def become_apikey_app(self, project_id):
```

Comment:
I don't understand this name.

Reply:
The `TestAppManager` (renamed from `TestAppMaker`) needs to be told what kind of app to make. We currently have a sample app which uses api keys to protect mutation methods, but I will be helping @kryzthov to add one which uses service account authorization.

This method mutates internal state of the `TestAppManager`, rather than create a new object, so I named the method `become_apikey_app` instead of `make_apikey_app`. Let me know if you have a good alternative, please!

Reply:
Can you just add a short comment, for future reference?

Reply:
Added.